### PR TITLE
Free Merchants - new NPC classes and a trading post [help needed/almost done]

### PR DESCRIPTION
**Pull request by Brytenwalda**
#### Summary
```SUMMARY:  Content "A garage trading post with a merchant and two guards."```

#### Purpose of change

I believe that the game needs more trading outposts where players can exchange items. We already have a faction to do it, why not expand it a bit?

Overall, this should add another level of plausibility to this world - a traveling merchant who brings various individuals - hunters, fishermen, scavengers, and other survivors together through trade.

#### Describe the solution

1. Two new NPC classes - a field merchant and a guard.
2. A new mapgen building - a copy of the existing garage, but with the merchant and two guards in it.
3. I added the new trader garage to the regional map settings file, with a reasonably low chance to spawn.

#### Problem

As of this moment, the game fails to recognize the new NPC classes that I've added to classes.json. I would greatly appreciate any help to make them work .

### CODE REVIEW

File: data/json/npcs/FM_TRADER.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    "mission": 3,
+    "chat": "TALK_TRADER",
+    "faction": "free_merchants"
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_TRADER",
+    "dynamic_line": {
+      "u_is_wearing": "badge_marshal",
+      "yes": "Marshal,  I am a member of the Free Merchants, care to trade?",
```
Comment (Qrox) [inline data/json/npcs/FM_TRADER.json]:
```suggestion
      "yes": "Marshal, I am a member of the Free Merchants, care to trade?",
```

File: data/json/npcs/FM_TRADER.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      "no": {
+        "u_male": "Good to see you, Sir. I am an authorised member of the Free Merchants, care to trade?",
```
Comment (Qrox) [inline data/json/npcs/FM_TRADER.json]:
```suggestion
        "u_male": "Good to see you, Sir.  I am an authorised member of the Free Merchants, care to trade?",
```

File: data/json/npcs/FM_TRADER.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        "no": "Good to see you, Ma'am. I am a member of the Free Merchants, care to trade?"
```
Comment (Qrox) [inline data/json/npcs/FM_TRADER.json]:
```suggestion
        "no": "Good to see you, Ma'am.  I am a member of the Free Merchants, care to trade?"
```

File: data/json/npcs/FM_TRADER.json [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    "responses": [
+      { "text": "Free Merchants?", "topic": "TALK_TRADER_WHAT" },
+      { "text": "What are you doing here?", "topic": "TALK_TRADER_WHY" },
+      { "text": "See you later.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_TRADER_WHAT",
+    "dynamic_line": "We are a loose organisation, if you can even call us one. Survivors need food, medicine, clothes, scrap - and we can provide it. I made myself known to the local hunters and scavengers - they sell what they can gather, and buy what they need. Win-win.",
```
Comment (Qrox) [inline data/json/npcs/FM_TRADER.json]:
```suggestion
    "dynamic_line": "We are a loose organisation, if you can even call us one.  Survivors need food, medicine, clothes, scrap - and we can provide it.  I made myself known to the local hunters and scavengers - they sell what they can gather, and buy what they need.  Win-win.",
```

File: data/json/npcs/FM_TRADER.json [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    "responses": [
+      { "text": "Cheers for the free trade!", "topic": "TALK_TRADER" },
+      { "text": "Nah, I don't need anything.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_TRADER_WHY",
+    "dynamic_line": "Got my own little shop here, survivors around know where to find me. When they gather a nice loot, they sell it, and when they're starving or bleeding, they come here to get what they need. Simple.",
```
Comment (Qrox) [inline data/json/npcs/FM_TRADER.json]:
```suggestion
    "dynamic_line": "Got my own little shop here, survivors around know where to find me.  When they gather a nice loot, they sell it, and when they're starving or bleeding, they come here to get what they need.  Simple.",
```

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/new-npcs-i-need-your-help/21800/1

**Comment by Night-Pryanik** [comment]
Your created roof (`1s_garage__trader_roof`, btw why there are two underscores between `garage` and `trader`?) won't spawn at all. You need to created a new entry with your ground floor and your roof in any fitting json in `\data\json\overmap\overmap_terrain\`. Also you need to create a new entry in `\data\json\overmap\multitile_city_buildings.json` for both of your ground floor and your roof.

**Pull request closed without merging**